### ISSUE TITLE
Distribute a default data/options.dist.php

### ISSUE THREAD

**Issue by nodiscc** [shaarli/Shaarli#166]
`data/options.dist.php` should contain a copy of all configurable settings (those listed in index.php) and their defaults. Anytime, if `data/options.php` doesn't exist, recreate it by copying `data/options.dist.php -> data/options.php`.

This prevents temptation to edit index.php directly and presents all available options to users directly in the config file.

**Comment by virtualtam** [comment]
this may become invalid/unnecessary after #255

**Comment by ArthurHoaro** [comment]
`config.php` now contains all settings, including credentials. If the file is deleted, `install()` will be called to recreate it with valid values. 

To prevent edition in `index.php`, we can add:

> // DO NOT FUCKING TOUCH THESE! Change them in /data/config.php.

I suggest we close this as wontfix.

**Comment by virtualtam** [comment]
Agreed for closing as *won't fix*

Regarding settings, there's a nice module to politely handle user mistakes: [ArnoldC](https://github.com/lhartikk/ArnoldC) :D

**Comment by virtualtam** [comment]
I've moved the configuration-related documentation to a [dedicated page](https://github.com/shaarli/Shaarli/wiki/Configuration), and replaced the available option list with a commented version of `data/config.php`

**Comment by ArthurHoaro** [comment]
:+1:

**Comment by nodiscc** [comment]
> config.php now contains all settings, including credentials. If the file is deleted, install() will be called to recreate it with valid values. 

That was the point of this issue. Thanks!
